dartagnan/src/main/java/com/dat3m/dartagnan/utils/visualization/ExecutionGraphVisualizer.java

```diff
@@ -131,7 +128,7 @@ private ExecutionGraphVisualizer addAllThreadPos(ExecutionModel model) {
 
     private ExecutionGraphVisualizer addThreadPo(Thread thread, ExecutionModel model) {
         List<EventData> threadEvents = model.getThreadEventsMap().get(thread)
-            .stream().filter(e -> e.isMemoryEvent()).collect(Collectors.toList());
+            .stream().filter(e -> e.isMemoryEvent() || e.isFence()).collect(Collectors.toList());
```

Comment:
You can simply check for `Tag.VISIBLE` I think. IIRC that was also the code before you changed it in a previous PR.